File: src/main/java/common/CommonUtils.java
package common;

import java.io.IOException;
import java.net.InetAddress;
import java.util.Random;


public class CommonUtils {

    public static final String GITHUB_ROOT = "https://raw.githubusercontent.com/yudong80/reactivejava/master/";

    public static final String ERROR_CODE = "-500";


    public static long startTime;

    public static void exampleStart() {
        startTime = System.currentTimeMillis();
    }

    public static void sleep(int millis) {
        try {
            Thread.sleep(millis);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public static void doSomething() {
        try {
            Thread.sleep(new Random().nextInt(100));
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public static String numberToAlphabet(long x) {
        return Character.toString(ALPHABET.charAt((int) x % ALPHABET.length()));
    }


    public static boolean isNetworkAvailable() {
        try {
            return InetAddress.getByName("www.google.com").isReachable(1000);
        } catch (IOException e) {
            Log.it("Network is not available");
        }
        return false;
    }

}

File: src/main/kotlin/common/CommonUtils_.kt
package common

import java.io.IOException
import java.net.InetAddress
import java.util.*


class CommonUtils_ {

    companion object {

        val GITHUB_ROOT = "https://raw.githubusercontent.com/yudong80/reactivejava/master/"

        val ERROR_CODE = "-500"


        var startTime: Long = 0

        fun exampleStart() {
            startTime = System.currentTimeMillis()
        }

        fun sleep(millis: Int) {
            Thread.sleep(millis.toLong())
        }


        fun doSomething() {
            Thread.sleep(Random().nextInt(100).toLong())
        }


        private val ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

        fun numberToAlphabet(x: Int): String =
                ALPHABET[x % ALPHABET.length].toString()


        fun isNetworkAvailable(): Boolean {
            try {
                return InetAddress.getByName("www.google.com").isReachable(1000)
            } catch (e: IOException) {
                Log.it("Network is not available")
            }

            return false
        }

    }

}
